include/oneapi/mkl/sparse_blas/detail/onemkl_sparse_blas_backends.hxx

```diff
@@ -37,6 +37,10 @@ set_csr_data(sycl::queue &queue, matrix_handle_t handle, intType num_rows, intTy
              intType nnz, index_base index, intType *row_ptr, intType *col_ind, fpType *val,
              const std::vector<sycl::event> &dependencies = {});
 
+ONEMKL_EXPORT sycl::event optimize_gemm(sycl::queue &queue, transpose transpose_A,
+                                        matrix_handle_t handle,
+                                        const std::vector<sycl::event> &dependencies = {});
+
```

Comment:
There are 2 `optimize_gemm()` APIs in the oneMKL spec v1.3. There's the one you have here, and there's another one that also has the dense matrix info in the arguments. See [optimize_gemm() in oneMKL spec](https://spec.oneapi.io/versions/1.3-provisional-rev-1/elements/oneMKL/source/domains/spblas/gemmoptimize.html#onemkl-sparse-optimize-gemm). Could you please add both versions in?

Reply:
Thank you, I didn't notice that. I've added it here https://github.com/oneapi-src/oneMKL/pull/403/commits/943c1f430f214787fc577e955f51662d05e18f30
Currently the tests will always call the 2 `optimize_gemm` versions. I did not feel the need to duplicate the number of tests for this.

Reply:
We definitely don't want users calling both versions of GEMM one after the other, that's not the intended use case of the APIs (although it would work, but be pointless and possibly less performant). You could use one version of optimize for buffer tests and another one for USM tests, that would be better I think.

Reply:
But these are unit tests, so both versions need to be tested. How about adding a flag and an additional `EXPECT_TRUEORSKIP()` test in your list of tests in the CPP files?
```cpp
if (optimize_flag == 1)      {optimize_gemm(/*version 1*/);}
else if (optimize_flag == 2) {optimize_gemm(/*version 2*/);}
else                         { /* skip call to optimize_gemm(); that must work too! */ }
```

Reply:
That's not something that we tested for gemv but sure. I have added this for gemv in https://github.com/oneapi-src/oneMKL/pull/403/commits/9fab428f7f6c3214f23710593914c64d03e14d43 and for gemm in https://github.com/oneapi-src/oneMKL/pull/403/commits/0441c3ee899823eb0587f729b3cc448dfe7c6a9c
I've also added a test that uses both optimize_gemm to be complete.

Reply:
Yeah, I missed out on this during my review. Thanks for adding it for GEMV too!